lib/client.js

```diff
@@ -1939,11 +1967,46 @@ MatrixClient.prototype.registerGuest = function(opts, callback) {
 };
 
 /**
- * Set a list of rooms the Guest is interested in receiving events from.
- * @param {String[]} roomIds A list of room IDs.
+ * Peek into a room and receive updates about the room. This only works if the
+ * history visibility for the room is world_readable.
+ * @param {String} roomId The room to attempt to peek into.
+ * @return {module:client.Promise} Resolves: Room object
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
  */
-MatrixClient.prototype.setGuestRooms = function(roomIds) {
-    this._guestRooms = roomIds;
+MatrixClient.prototype.peekInRoom = function(roomId) {
+    if (this._peekSync) {
+        this._peekSync.stopPeeking();
+    }
+    this._peekSync = new SyncApi(this);
+    return this._peekSync.peek(roomId);
+};
+
+/**
+ * Set r/w flags for guest access in a room.
+ * @param {string} roomId The room to configure guest access in.
+ * @param {Object} opts Options
+ * @param {boolean} opts.allowJoin True to allow guests to join this room. This
+ * implicitly gives guests write access. If false or not given, guests are
+ * explicitly forbidden from joining the room.
+ * @param {boolean} opts.allowRead True to set history visibility to
+ * be world_readable. This gives guests read access *from this point forward*.
+ * If false or not given, history visibility is not modified.
+ * @return {module:client.Promise} Resolves: TODO
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.setGuestAccess = function(roomId, opts) {
+    var writePromise = this.sendStateEvent(roomId, "m.room.guest_access", {
+        guest_access: opts.allowJoin ? "can_join" : "forbidden"
+    });
+
+    var readPromise = q();
+    if (opts.allowRead) {
+        readPromise = this.sendStateEvent(roomId, "m.room.history_visibility", {
+            history_visibility: "world_readable"
+        });
+    }
```

Comment:
> Do we not want to allow people turn world readability off again?

We don't reasonably know what the previous history visibility was to set it back to if they decide to turn it off.

>  is there a reason for these params being in a object rather than... parameters?

Because this makes it much more extensible when inevitably we add in more params. I wish I did this on more things earlier (e.g. `MatrixClient.register()`).